### ISSUE TITLE
Investigate keyboard and mouse selection of search results

### ISSUE THREAD

**Issue by owennw** [owennw/OpenFinD3FC#297]
Now that the keyboard can be used to change selected search result the user experience might not be as expected.

By hovering the mouse over one search result, then using the keyboard to select others two separate results can appear highlighting, and then moving the mouse over others won't reduce that down to 1. 

Should selecting with the keyboard act as a pure 'selection', i.e. not change the displayed stock until a mouse click or enter button has been pressed? And how should this work with the mouse?

**Comment by akhmorgan** [comment]
I'd say the behaviour at present is pretty self-explanatory to my mind. If anything I'd say that we should change the hover (mouse over) state of the search results to distinguish it from a selected result. So that it is clear that keyboard changes selected result, whereas mouse over doesn't until you click, which makes sense to me.

**Comment by owennw** [comment]
Interesting. Would you apply the same styling rules to the favourite cards? Separating selected from hovered?

**Comment by akhmorgan** [comment]
I suppose that would make sense for the sake of consistency, so yes.

**Comment by WPFerg** [comment]
I agree. Perhaps a 50% faded background colour when hovering?

**Comment by akhmorgan** [comment]
Yes, something like that should work. We should maybe run any ideas past Rui before proceeding though.

**Comment by owennw** [comment]
Definitely.

**Comment by mic-smith** [comment]
@rmarcalo are you happy with the change to separate the selected and hover state?

**Comment by rmarcalo** [comment]
I believe the proposed solution might work. A 50% opacity for hover might be hard to notice in some monitors but it's definitely worth testing and tweaking as required.

**Comment by mic-smith** [comment]
It's pretty subtle on the favourites. Looks like:
![image](https://cloud.githubusercontent.com/assets/17762186/13841328/2dac7b8e-ec20-11e5-82d3-334c3dd5c193.png)

![image](https://cloud.githubusercontent.com/assets/17762186/13841331/32384a66-ec20-11e5-8464-f9c2941f275e.png)

**Comment by owennw** [comment]
@mic-smith I can't see the difference on the favourites! :/ Might be my monitor/contrast/eyes though.

**Comment by mic-smith** [comment]
@owennw Yep it's not the most obvious. Going much lighter went too close to the normal background colour - at least on my monitor.
I tried a 50% alpha version which ended up with a green hint from the drop area below. This is more prominent to my eyes and similar to the hover effect on the collapsed favourites. But then ends up not being as consistent with the hover effect on the search in terms of just darkening the selection. 

![image](https://cloud.githubusercontent.com/assets/17762186/13845199/41686622-ec36-11e5-890c-9cf5a8c9c098.png)

@rmarcalo Do you have any preference?

**Comment by rmarcalo** [comment]
I think the green hover could be applied to the search too and would look quite interesting.

**Comment by mic-smith** [comment]
This is the hover set to the same colour as the collapsed sidebar section hover. It's greener now on the favourites as  previously there was an alpha.

![image](https://cloud.githubusercontent.com/assets/17762186/13848355/44d79cac-ec49-11e5-8037-921474512611.png)

![image](https://cloud.githubusercontent.com/assets/17762186/13848357/46874f02-ec49-11e5-9e01-8e576e774bee.png)

@rmarcalo This OK, or prefer any of the previous versions?

**Comment by rmarcalo** [comment]
Hi, I much prefer the green on the previous image than the new one on the favourites bar, could we use that one instead?

**Comment by mic-smith** [comment]
Sure. And same colour for the search?

**Comment by rmarcalo** [comment]
Yes, I think it will still fit nicely.

**Comment by mic-smith** [comment]
@rmarcalo Thanks. For reference this is what the search looks like:

![image](https://cloud.githubusercontent.com/assets/17762186/13851348/b533b024-ec55-11e5-89bb-31d027d10904.png)

**Comment by rmarcalo** [comment]
I was starting to have a look at the overall keyboard navigation for the application. Would it be possible to provide me with a list of all keyboard keys (and what they do in each situation) used so far?

**Comment by rmarcalo** [comment]
I've attached my ideal designs for the keyboard shortcuts and how to navigate through the main parts of the interface. Usually keyboard shortcuts are meant to replace quick interactions, rather than serve as a replacement for the mouse altogether, so some features are still only accessible using the latter.  

In the designs I've attached below, there is a "hidden" keyboard shortcuts menu for the advanced user, accessible by pressing ? (or shift+? in the uk)  which is a web standard (used for google products,github, jira, etc). 

Without knowledge of these specific keys (of which there aren't many), the user can still use "Tab" , the arrows and "Enter" intuitively.

![keyboard navigation](https://cloud.githubusercontent.com/assets/13066377/14148694/6cffc7e6-f698-11e5-9bd5-21716565bb82.png)
